lib/sles4sap/sdaf_library.pm

```diff
@@ -445,6 +553,105 @@ sub get_tfvars_path {
+    my (%args) = @_;
+    my %tfvars_os_variable = (
+        deployer => 'deployer_parameter_file',
+        sap_system => 'sap_system_parameter_file',
+        workload_zone => 'workload_zone_parameter_file',
+        library => 'library_parameter_file'
+    );
+    my %tfvars_template_url = (
+        deployer => data_url('sles4sap/sdaf/DEPLOYER.tfvars'),
+        sap_system => data_url('sles4sap/sdaf/SAP_SYSTEM.tfvars'),
+        workload_zone => data_url('sles4sap/sdaf/WORKLOAD_ZONE.tfvars'),
+        library => data_url('sles4sap/sdaf/LIBRARY.tfvars')
+    );
+    croak 'Deplyoment type not specified' unless $args{deployment_type};
+    croak "Unknown deployment type: $args{deployment_type}" unless $tfvars_os_variable{$args{deployment_type}};
+
+    my $tfvars_file = get_os_variable($tfvars_os_variable{$args{deployment_type}});
+    my $retrieve_tfvars_cmd = join(' ', 'curl', '-v', '-fL', $tfvars_template_url{$args{deployment_type}}, '-o', $tfvars_file);
+
+    assert_script_run($retrieve_tfvars_cmd);
+    assert_script_run("test -f $tfvars_file");
+    replace_tfvars_variables($tfvars_file);
+    upload_logs($tfvars_file, log_name => "$args{deployment_type}.tfvars");
+    return $tfvars_file;
+}
+
+=head2 replace_tfvars_variables
+
+    replace_tfvars_variables();
+
+B<$deployment_type>: Type of the deployment (workload_zone, sap_system, library... etc)
+
+Replaces placeholder pattern B<%OPENQA_VARIABLE%> with corresponding OpenQA variable value.
+If OpenQA variable is not set, placeholder is replaced with empty value.
+
+=cut
+
+sub replace_tfvars_variables {
+    my ($tfvars_file) = @_;
+    croak 'Variable "$tfvars_file" undefined' unless defined($tfvars_file);
+    my @variables = qw(SDAF_ENV_CODE SDAF_LOCATION SDAF_RESOURCE_GROUP SDAF_VNET_CODE SAP_SID);
+    my %to_replace = map { '%' . $_ . '%' => get_var($_, '') } @variables;
+    file_content_replace($tfvars_file, %to_replace);
+}
+
+=head2 sdaf_deploy_workload_zone
+
+    sdaf_deploy_workload_zone();
+
+Executes SDAF workload zone deployment. SDAF relies on OS env variables therefore those are passed as cmd args as well.
+Definitely keep I<--spn-secret> set as a reference to OS variable I<${ARM_CLIENT_SECRET}>, otherwise password will
+be shown in openQA output log in plaintext.
+L<https://learn.microsoft.com/en-us/azure/sap/automation/deploy-workload-zone?tabs=linux#deploy-the-sap-workload-zone>
+
+=cut
+
+sub sdaf_deploy_workload_zone {
+    my ($tfvars_filename, $tfvars_path) = fileparse(get_os_variable('workload_zone_parameter_file'));
+    # Variable is specific to each deployment type and will be changed during the course of whole deployment process.
+    # It is used by SDAF internally, so keep it set in OS env
+    set_os_variable('parameterFile', $tfvars_filename);
+
+    # SDAF has to be executed from the profile directory
+    assert_script_run("cd $tfvars_path");
+    my $deploy_command = join(' ', '$SAP_AUTOMATION_REPO_PATH/deploy/scripts/install_workloadzone.sh',
```

Comment:
Other places are composing deploy/scripts/ path in a different way https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19048/files#diff-a7b59ba45f396aa9b361d43f76fb4f6669a4b42c5d0e1dc952bfa526ec508934R795